test/unit/module/core/simd_cast.cpp

```diff
@@ -78,35 +77,50 @@ simd_cast_test_one_way()
     {
       TTS_EQUAL(expected_logical_bytes.get(i), actual_logical_bytes.get(i));
     }
-    return;
   }
+  else
+  {
+    constexpr std::size_t t_size = sizeof(eve::element_type_t<T>) * T::size();
+    constexpr std::size_t u_size = sizeof(eve::element_type_t<U>) * U::size();
 
-  constexpr std::size_t t_size = sizeof(eve::element_type_t<T>) * T::size();
-  constexpr std::size_t u_size = sizeof(eve::element_type_t<U>) * U::size();
-
-  std::array<std::uint8_t, t_size> expected_buf;
-  std::array<std::uint8_t, u_size> actual_buf;
+    std::array<std::uint8_t, t_size> expected_buf;
+    std::array<std::uint8_t, u_size> actual_buf;
 
-  std::memcpy(expected_buf.data(), &x, t_size);
-  std::memcpy(actual_buf.data(), &actual, u_size);
+    std::memcpy(expected_buf.data(), &x, t_size);
+    std::memcpy(actual_buf.data(), &actual, u_size);
 
-  std::size_t min_size = std::min(t_size, u_size);
+    std::size_t min_size = std::min(t_size, u_size);
 
-  TTS_EQUAL(std::vector(expected_buf.data(), expected_buf.data() + min_size),
-            std::vector(actual_buf.data(), actual_buf.data() + min_size),
-            REQUIRED)
-      << "from: " << tts::typename_<T> << " to: " << tts::typename_<U>;
+    TTS_EQUAL(std::vector(expected_buf.data(), expected_buf.data() + min_size),
+              std::vector(actual_buf.data(), actual_buf.data() + min_size),
+              REQUIRED)
+        << "from: " << tts::typename_<T> << " to: " << tts::typename_<U>;
+  }
 }
 
 template<typename T, typename U>
 void
 simd_cast_test()
 {
-  // saving compile time a bit
-  if constexpr( T::size() * 2 <= U::size() )
+  // for risc-v memory consumption is too big,
```

Comment:
Note: still a quite a lot of memory is used.

To decrease consumption more, I can remove these types checks for RISC-V: 

https://github.com/jfalcou/eve/blob/7e784587c274c57c63505d74e8b68380c83bff60/test/unit/module/core/simd_cast.cpp#L129C9-L129C84

Reply:
It occurs we anyway have a large memory consumption. Changed to test only float types with expected and fundamental cardinals. Hope this will be enough.

Reply:
> Changed to test only float types with expected and fundamental cardinals. Hope this will be enough.

Changed to check only expected and fundamental cardinals with casts to `unsigned char` wides. We will test lmul extend/lower functionality while casting to `wide<std::uint8_t, sizeof(T) * 2>`, and same sizes while casting to `wide<std::uint8_t, sizeof(T)>`.